encrypt.go

```diff
@@ -0,0 +1,77 @@
+package dht
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha256"
+	"errors"
+
+	"github.com/multiformats/go-multihash"
+)
+
+const (
+	nonceSize             = 12
+	keySize               = 32
+	encryptedPeerIDLength = 66
+)
+
+var errInvalidKeySize = errors.New("key size must be 32 bytes")
+
+func encryptAES(plaintext, key []byte) ([]byte, error) {
+	aesgcm, err := newAESGCM(key)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce := make([]byte, nonceSize)
+	_, err = rand.Read(nonce)
+	if err != nil {
+		return nil, err
+	}
+
+	ct := aesgcm.Seal(nil, nonce, plaintext, nil)
+	return append(nonce, ct...), nil
```

Comment:
We probably need to append encryption algorithm here too so that the encryption function rotation doesn't require coordinated client update. I.e. for example `AESGCM || nonce || enc(payload)`. That would also allow to have different encryption functions in the same DHT.

Reply:
good point, will add!

Reply:
This is going to add 6 bytes to every encrypted value. Can we use a varint as code for AESGSM, then update the spec to document it?

Reply:
@masih do you have a pointer to the AESGCM varint?

Reply:
I believe we will have to define this mask ourselves and then document in the spec what that number means

Reply:
Right. We'd nee to define those in the spec ourselves. 

This can also be a new multicodec. I recommend picking a varint that's not reserved already to have the option of adding it to the multicodec table later on.